src/firecracker/src/main.rs

```diff
@@ -40,6 +40,13 @@ fn main() {
         process::exit(i32::from(vmm::FC_EXIT_CODE_GENERIC_ERROR));
     }
 
+    // Make stdout non blocking.
+    let flags = unsafe { libc::fcntl(libc::STDOUT_FILENO, libc::F_GETFL, 0) };
```

Comment:
According to the [fcntl man page](https://linux.die.net/man/2/fcntl), the only error we could get from this particular `fcntl` call would be `EBADF`. This only happens when `STDOUT_FILENO` is not an open FD, but this is never the case in Firecracker.

Correct me if I'm wrong, but I think even when running in a daemonized jailer, the FD still isn't closed, but redirected to `/dev/null`.

Reply:
I don't think it states anywhere in the man page that this is the only error that can be returned for the `F_GETFL` command.

It says that it can return -1, setting the appropriate errno corresponding to the error type. We can only presume based on heuristics that this is the only error we could encounter (and I don't think that is right).

The added cost of checking the return value is sooo tiny we should just do it to be safe and future-proofed

Reply:
When looking to the possible error codes returned, no error code situation fits ours aside from `EBADF`. This is what I meant earlier, it is indeed not stated explicitly.

> The added cost of checking the return value is sooo tiny we should just do it to be safe and future-proofed

I don't entirely agree with this. I think we should not add a check with no particular reason. While in this case the cost is minimal, I would like us to be consistent in our approach with these checks and add them if we either need to cover a code path right now or we see a possibility in the near future for a change in our assumptions. Neither of these is the case with this issue.

As the cost of this is minimal and I seem to be on the wrong side of this issue, I will add the check.

Reply:
I think we should **always** check syscall return codes. Even if in some cases _we know_ they can't fail. It's just an easy thing to do that will always keep us on the safe side.

I don't think there are any code paths hot enough to worry about a potentially useless error check, so I don't think we should take on the extra complexity and effort to validate _when we need to check and when we don't_. Just always do it 😄